queen.go

```diff
@@ -224,76 +156,103 @@ func (q *Queen) Run(ctx context.Context) error {
 	}
 }
 
-func (q *Queen) consumeAntsLogs(ctx context.Context) {
-	requests := make([]models.RequestsDenormalized, 0, q.resolveBatchSize)
+func (q *Queen) consumeAntsEvents(ctx context.Context) {
+	requests := make([]*db.Request, 0, q.cfg.BatchSize)
+
 	// bulk insert for every batch size or N seconds, whichever comes first
-	ticker := time.NewTicker(time.Duration(q.resolveBatchTime) * time.Second)
+	ticker := time.NewTicker(q.cfg.BatchTime)
 	defer ticker.Stop()
 
 	for {
 		select {
-
 		case <-ctx.Done():
 			logger.Debugln("Gracefully shutting down ants...")
 			logger.Debugln("Number of requests remaining to be inserted:", len(requests))
+
 			if len(requests) > 0 {
-				err := db.BulkInsertRequests(context.Background(), q.dbc.Handler, requests)
-				if err != nil {
-					logger.Fatalf("Error inserting remaining requests: %v", err)
+				if err := q.clickhouseClient.BulkInsertRequests(ctx, requests); err != nil {
+					logger.Errorf("Error inserting requests: %v", err)
 				}
+				requests = requests[:0]
 			}
 			return
 
-		case log := <-q.antsLogs:
-			reqType := kadpb.Message_MessageType(log.Type).String()
-			maddrs := q.peerstore.Addrs(log.Requester)
-			var agent string
-			peerstoreAgent, err := q.peerstore.Get(log.Requester, "AgentVersion")
-			if err != nil {
-				if peerstoreAgent, ok := q.agentsCache.Get(log.Requester.String()); ok {
-					agent = peerstoreAgent.(string)
-				} else {
-					agent = ""
+		case evt := <-q.antsEvents:
+
+			// transform multi addresses
+			maddrStrs := make([]string, len(evt.Maddrs))
+			for i, maddr := range evt.Maddrs {
```

Comment:
my reasoning here was that we can be sure that the peerstore is populated with data while we have an open stream which is the case when we're handling a DHT request.

Doing this outside of that scope makes us sensitive to timeouts and the GC logic of the peer store. For example, I don't know how the peer store prunes data when a peer disconnects. So if we have a slow database insert, for whatever reason, and in the meantime the peer disconnects, is the data still available in the peer store? It could still be there but that could also change with the next go-libp2p update, so my idea was to not be coupled to that logic. The same applies to the protocols and agent version data.

Reply:
Sounds good, I am now convinced